File: exercism/python/nth-prime/nth_prime.py
# https://exercism.io/my/solutions/9483c47c57bb449cb6395acec6256dae

from math import sqrt


def is_prime(number):
    if number < 2:
        return False

    for i in range(2, int(sqrt(number)) + 1):
        if number % i == 0:
            return False
    return True


def prime(nth):
    if nth < 1:
        raise ValueError("Wrong argument.")
    if nth == 1:
        return 2

    count = 1
    n = 1
    while count < nth:
        n += 2
        if is_prime(n):
            count += 1
    return n

File: exercism/go/nth-prime/nth_prime.go
// https://exercism.io/my/solutions/c5526f43f30b49e28c14137271b2bbde

package prime

// isPrime determines whether a number is prime or not
func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	for i := 2; i*i <= n; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

// Nth returns the n-th primer number
func Nth(nth int) (int, bool) {
	if nth < 1 {
		return 0, false
	}
	if nth == 1 {
		return 2, true
	}
	count := 1
	for n := 3; ; n += 2 {
		if isPrime(n) {
			count++
			if count == nth {
				return n, true
			}
		}
	}
}
